src/ClientContext.tsx

```diff
@@ -107,8 +119,18 @@ export const ClientProvider: FC<Props> = ({ children }) => {
     client: undefined,
     userName: null,
     error: undefined,
+    disconnected: false,
   });
 
+  const onSync = (state: SyncState, _old: SyncState, data: ISyncStateData) => {
+    setState((currentState) => {
+      return {
+        ...currentState,
+        disconnected: isDisconnected(state, data),
+      };
+    });
+  };
```

Comment:
I'm slightly concerned that this could cause the entire component tree to re-render on every sync, because it sets the state to a new object every time (and React [doesn't perform a shallow comparison on state updates in functional components, but rather an `Object.is` comparison](https://react.dev/reference/react/useState#setstate-caveats)). Something like this would avoid that issue:

```suggestion
  const onSync = (state: SyncState, _old: SyncState, data: ISyncStateData) => {
    setState((currentState) => {
      const disconnected = isDisconnected(state, data)
      return disconnected === currentState.disconnected
        ? currentState
        : { ...currentState, disconnected };
    });
  };
```

Reply:
Thanks!